api/integrations/slack.js

```diff
@@ -0,0 +1,77 @@
+var q = require('q');
+var Slack = require('slack-node');
+
+module.exports = function () {
+    global.app.post('/integration/slack', function (req, res) {
+
+        var appId = req.body.appid;
+        var appkey = req.body.appkey;
+        var event_type = req.body.event_type;
+        var user = req.body.user;
+
+        console.log(appId, appkey, event_type);
+
+        var response = { appid: req.body.appid, status: "", user: user }
+
+        global.appService.isKeyValid(appId, appkey).then(function (isKeyValid) {
+            if (isKeyValid) {
+
+                if (global.mongoDisconnected) {
+                    return res.status(500).send('Storage Services are temporarily down');
+                }
+
+                var slack = new Slack();
+                global.appService.getAllSettings(appId).then(function (settings) {
+                    settings
+                    var slackSettings;
+                    settings.forEach(function(element) {
+                        if(element.category == "integrations"){
+                            slackSettings = element.settings.slack;
+                        }
+                    }, this);
+                    slack.setWebhook(slackSettings.webhook_url);
+                    console.log(slackSettings, "Settings")
+                    if (slackSettings.enabled === true) {
```

Comment:
we can have many custom events, not just login and signup , so modify the API to accommodate those as well , make a generic slack webhook , which takes in event type and sends it slack with relevant data.